### ISSUE TITLE
Manual snupkg Push from Deployment Environment Not Working - "No packages were published"

### ISSUE THREAD

**Issue by lordmilko** [appveyor/ci#3576]
I have an Environment for my [project](https://ci.appveyor.com/project/lordmilko/prtgapi) in AppVeyor for manually pushing my artifacts to NuGet. There are two artifacts that should be published: my *.nupkg file, and my *.snupkg file

About a year ago, this worked fine with the symbol server specified as simply `nuget.org`

```
Downloading artifact PrtgAPI.0.9.11.nupkg to temporary location...OK
Publishing PrtgAPI.0.9.11.nupkg to https://www.nuget.org/api/v2/package...OK
Downloading artifact PrtgAPI.0.9.11.snupkg to temporary location...OK
Publishing PrtgAPI.0.9.11.snupkg to https://www.nuget.org/api/v2/symbolpackage...OK
Total packages published: 2
```

Today when I tried to deploy however I got an error - it seems that the symbol server `nuget.org` is no longer considered valid

```
Downloading artifact PrtgAPI.0.9.14.nupkg to temporary location...OK
Publishing PrtgAPI.0.9.14.nupkg to https://www.nuget.org/api/v2/package...OK
Downloading artifact PrtgAPI.0.9.14.snupkg to temporary location...OK
Invalid NuGet source URL: nuget.org
```

Perhaps Appveyor requires a more proper URL now; that's fine. When I try more proper URLs like `https://www.nuget.org/api/v2/symbolpackage` or `https://www.nuget.org` though I get the following

```
Downloading artifact PrtgAPI.0.9.14.nupkg to temporary location...OK
Publishing PrtgAPI.0.9.14.nupkg to https://www.nuget.org/api/v2/package...Skipped (A package with ID 'PrtgAPI' and version '0.9.14' already exists and cannot be modified.)
Downloading artifact PrtgAPI.0.9.14.snupkg to temporary location...OK
No packages were pushed.
```
In the case of the *.nupkg file, obviously that's being skipped because I'm now reattempting to push version 0.9.14. But in the case of the *.snupkg file, I don't understand why it's randomly deciding to skip it. It doesn't give me an error message or anything

According to [Microsoft's documentation](https://github.com/NuGet/docs.microsoft.com-nuget/blob/master/docs/create-packages/Symbol-Packages-snupkg.md), publishing snupkg's is apparently now only supported on the v3 API endpoint. And sure enough, I was able to manually publish my *.snupkg to nuget.org by specifying this v3 URL; specifying the `-source` as `https://www.nuget.org` or `https://www.nuget.org/api/v2/symbolpackage` did not work

```
Z:\>dotnet nuget push *.snupkg -s https://api.nuget.org/v3/index.json
Pushing PrtgAPI.0.9.14.snupkg to 'https://www.nuget.org/api/v2/symbolpackage'...

  PUT https://www.nuget.org/api/v2/symbolpackage/
  Created https://www.nuget.org/api/v2/symbolpackage/ 3425ms
Your package was pushed.
```

Even with my AppVeyor environment set to this new v3 URL however, it still wouldn't attempt to push the package - it just gives me "No packages were pushed". I'm not sure if that's just because it failed pushing the *.snupkg at the time it pushed the *.nupkg, so it's never going to work for this release version (but it may for the next), or there is simply something fundamentally wrong with the URL I'm specifying for the symbol server, and if I only corrected that it would begin working again.

Are you able to advise what needs to be done to get AppVeyor to push *.snupkg files along with *.nupkg files in a manual Deployment Environment configuration, or otherwise advise the extent to which AppVeyor may have a bug with regards to processing *.snupkg packages?

**Comment by FeodorFitsner** [comment]
Here is test project publishing both `.nuget` and `.snupkg` files: https://ci.appveyor.com/project/appveyor-tests/snupkgtest/builds/36536075

[`NuGet` deployment section](https://github.com/appveyor-tests/SnupkgTest/blob/74f0dc0a9278ae4cdbd272f25ea2b710f32d0c45/appveyor.yml) in `appveyor.yml` is just:

```yaml
deploy:
- provider: NuGet
  api_key:
    secure: <secure>
```

Hope that helps.

**Comment by lordmilko** [comment]
Hi @FeodorFitsner,

This project automatically publishes its artifacts to NuGet as part of the build process

```
Checking is snupkg files were created...
Deploying using NuGet provider
Publishing MySnupkgTestNetStandard.1.0.17.nupkg to https://www.nuget.org/api/v2/package...OK
Publishing MySnupkgTestNetFramework.1.0.17.nupkg to https://www.nuget.org/api/v2/package...OK
Publishing MySnupkgTestNetStandard.1.0.17.snupkg to https://www.nuget.org/api/v2/symbolpackage...OK
Publishing MySnupkgTestNetFramework.1.0.17.snupkg to https://www.nuget.org/api/v2/symbolpackage...OK
Total packages published: 4
Build success
```

In my scenario, I do not want to automatically publish artifacts, I want to manually publish artifacts when I am ready for release, which I do by going to the Deployments tab of my project and manually doing a deployment from an Environment I have configured for deploying to NuGet. Deploying snupkg files in this way used to work, but now it no longer seems to

**Comment by FeodorFitsner** [comment]
Publishing via "NuGet" environment works too:

![image](https://user-images.githubusercontent.com/5041459/100411875-c90f1a00-3027-11eb-86ff-c56ecf067ab0.png)

Environment settings:

![image](https://user-images.githubusercontent.com/5041459/100411905-e17f3480-3027-11eb-9561-639ed79eedbb.png)

**Comment by lordmilko** [comment]
The Symbol Server URL says it uses SymbolSource.org if no symbol server is specified, however I can see from your screenshots that no symbol source URL is specified and yet its somehow using nuget.org for symbols instead.

So are you saying that is in fact no longer the case that symbols will go to SymbolSource.org, or *.symbols.nupkg files will go to SymbolSource.org but *.snupkg files will go to nuget.org instead?

**Comment by FeodorFitsner** [comment]
Right, SymbolSource.org has been dead for a long time. Need to fix UI.
